spotlight/factorization/explicit.py

```diff
@@ -80,7 +82,7 @@ def __init__(self,
         self._l2 = l2
         self._use_cuda = use_cuda
         self._sparse = sparse
-        self._optimizer = None
+        self._optimizer = optimizer
```

Comment:
One caveat: we should store this under a different attribute here (`self._optimizer_func`?). If we don't do that, it'll be impossible to call `fit` twice, as it will try to use this attribute (which after the first `fit` call is not a function any more) to instantiate the optimizer.